build/template-OneBranch-CI-libsandsamples.yaml

```diff
@@ -21,27 +22,45 @@ steps:
   inputs:
     version: 6.x
 
-- task: DotNetCoreCLI@2
-  displayName: 'dotnet workload restore for Desktop project'
-  inputs:
-    command: 'custom'
-    custom: 'workload'
-    arguments: 'restore microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client.Desktop\Microsoft.Identity.Client.Desktop.csproj'
+- ${{ if eq(parameters.Mode, 'build') }}:
+  - task: DotNetCoreCLI@2
+    displayName: 'dotnet workload restore for Desktop project - Build'
+    inputs:
+      command: 'custom'
+      custom: 'workload'
+      arguments: 'restore $(Build.SourcesDirectory)/$(MsalSourceDir)src/client/Microsoft.Identity.Client.Desktop/Microsoft.Identity.Client.Desktop.csproj'
```

Comment:
Can you figure out a solution where you don't need to duplicate build tasks? You can have a task where you compute the source directory https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#understand-variable-syntax